### PR TITLE
Recipeapp frontend

### PR DESCRIPTION
**Pull request by afti-githobo**
## Description

Implements frontend for recipeapp.

Don't even think about merging this right now - it's nowhere near done, and it kinda sucks besides.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by afti-githobo** [comment]
I pushed the current state of the api2 stuff to this branch like a dolt w/o thinking, so I guess this is just a big omnibus PR now, which... sucks. Whatever.

Okay, so we have a WIP version of a higher-level API right now, anyway, and some minor reorganization to hide away some of the lower-level stuff in that new `low_level` import for code that actually wants it. This is still very much under construction - one thing I'd really like to be able to do since we're encapsulating all the native types anyhow is give datetime objects instead of block heights in the higher-level stuff. That seems doable but I have no idea how to do it.

What we *don't* have and I've been fretting about today: I really wanted to implement the desktop wallet-interface-to-pylonsd shim as a part of the SDK, but I'm gonna be honest, I'm getting the impression that's gonna be much more work than we thought it'd be. Notably: the uni_links stuff is embedded p. deeply into the sdk implementation, which seems... problematic? I don't know right now if there actually *is* a good way to shim this and I've got more reading to do tomorrow, so that fucking sucks.

It might be good to just abstract out some of the mobile-specific stuff? But that's more scope creep, so that's... fun. Scope creep is probably *inescapable* though, whether it's refactoring the SDK codebase to make the desired shim viable or writing a stub wallet... assuming we can even _write_ a client that interacts w/ the unilinks stuff on desktop; idk.

Anyway, this all sucks, but I'm feeling good about the API itself, even if it's not quite done yet.

**Comment by afti-githobo** [comment]
oh, one more thing i wanna drop here from today - this exists (https://github.com/leanflutter/uni_links_desktop) which seems promising, tho it would def. mean we have to have a whole stub wallet.

**Comment by MikeSofaer** [comment]
i would like to get @faddat's input on a sub wallet approach.

File: recipeapp/test/widget_test.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// gestures. You can also use WidgetTester to find child widgets in the widget
+// tree, read text, and verify that the values of widget properties are correct.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:recipeapp/main.dart';
+
+void main() {
+  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
```
Comment (kjawadDeveloper2) [inline recipeapp/test/widget_test.dart:14]: Remove this test. It will cause the CI to fail.

File: recipeapp/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+name: recipeapp
+description: App what runs recipes
+
+# The following line prevents the package from being accidentally published to
+# pub.dev using `flutter pub publish`. This is preferred for private packages.
+publish_to: 'none' # Remove this line if you wish to publish to pub.dev
+
+# The following defines the version and build number for your application.
```
Comment (kjawadDeveloper2) [inline recipeapp/pubspec.yaml:8]: Remove comments.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Comment by kjawadDeveloper2** [comment]
Did you checked how your these changes will effect the easel , wallet.

**Comment by kjawadDeveloper2** [comment]
Kindly check issues with analysis and test

**Comment by afti-githobo** [comment]
> Kindly check issues with analysis and test

this is a draft; I'm aware of these and they'll be corrected before it's merged, but it's not going to be merged yet.

re: changes affecting easel/wallet - anything using the old API will have to import low_level; that's it. I'll take care of that before merge, again.

What I actually need here are just, like - thoughts on what I was proposing w/ writing a stub wallet that sits between desktop applications and pylonsd and does the bare minimum of data processing to resolve unilinks. Notes on the behavior of the systems it's interacting with, concerns, potential pitfalls, etc.

**Pull request closed without merging**